scripts/zones/Abyssea-Konschtat/npcs/Cruor_Prospector.lua

```diff
@@ -94,14 +96,21 @@ local prospectorItems =
     },
 }
 
+-- Trust Alter Ego Extravaganza
+local cipher = 0
+local enable = xi.settings.main.ENABLE_TRUST_ALTER_EGO_EXTRAVAGANZA
+if enable == 1 or enable == 3 then -- 1=Summer/NY campaign 3=All campaigns
+    cipher = 1
+end
```

Comment:
Personal preference here, but I really don't like variables defined and logic outside functions.
You could create a function like "eventCheck" and place it there OR you could place all this logic inside entity.onTrigger